### PR TITLE
EVG-2499 Add button to project admin page to trigger "catchup run"

### PR DESCRIPTION
**Pull request by richardsamuels**
Implemented as a checkbox in the ui admin page. It only shows up when the webhook is setup and the repository is setup to track push events.

### CODE REVIEW

File: service/project.go [hunk 1]
```diff
@@ -236,6 +238,15 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if projectVars.GithubHookID != 0 && projectRef.TracksPushEvents &&
+		responseRef.ForceRepotrackerRun {
+		job := units.NewRepotrackerJob(util.RandomString(), projectRef.Identifier)
```
Comment (tychoish) [inline service/project.go]: can we have a better identifier here? 

there's also jobs.GetNumber() which provides a monotonically increasing job id, which might be useful.

Comment (richardsamuels) [reply service/project.go]: done

File: service/project.go [hunk 2]
```diff
@@ -236,6 +239,15 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if projectVars.GithubHookID != 0 && projectRef.TracksPushEvents &&
+		responseRef.ForceRepotrackerRun {
+		j := units.NewRepotrackerJob(fmt.Sprintf("%d", job.GetNumber()), projectRef.Identifier)
```
Comment (tychoish) [inline service/project.go]: can this be "ui-triggered-job-%d"

Comment (richardsamuels) [reply service/project.go]: will do

Comment (richardsamuels) [reply service/project.go]: done; also changed the other repo tracker job to have a prefix.

**Pull request merged**